common/src/main/java/com/wynntils/gui/screens/ChatTabEditingScreen.java

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.gui.screens;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.chat.RecipientType;
+import com.wynntils.core.chat.tabs.ChatTab;
+import com.wynntils.core.managers.Managers;
+import com.wynntils.core.managers.Models;
+import com.wynntils.features.user.ChatTabsFeature;
+import com.wynntils.gui.render.FontRenderer;
+import com.wynntils.gui.render.HorizontalAlignment;
+import com.wynntils.gui.render.VerticalAlignment;
+import com.wynntils.gui.widgets.TextInputBoxWidget;
+import com.wynntils.mc.objects.CommonColors;
+import com.wynntils.mc.utils.McUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.components.Checkbox;
+import net.minecraft.client.gui.screens.ChatScreen;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.resources.language.I18n;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.network.chat.TranslatableComponent;
+import org.lwjgl.glfw.GLFW;
+
+public class ChatTabEditingScreen extends Screen implements TextboxScreen {
+    private TextInputBoxWidget focusedTextInput;
+
+    private TextInputBoxWidget nameInput;
+    private TextInputBoxWidget autoCommandInput;
+    private TextInputBoxWidget orderInput;
+    private final List<Checkbox> recipientTypeBoxes = new ArrayList<>();
+    private TextInputBoxWidget filterRegexInput;
+    private Checkbox consumingCheckbox;
+
+    private Button saveButton;
+    private Button deleteButton;
+
+    private boolean firstSetup;
+    private final ChatTab edited;
+
+    public ChatTabEditingScreen() {
+        this(null);
+    }
+
+    public ChatTabEditingScreen(ChatTab tab) {
```

Comment:
To merge in the new wrapper, make this private and add a `create` factory method.

Reply:
Sorry, there were two constructors. Make both private and make two factory methods.